src/main/java/org/amahi/anywhere/fragment/ServerFilesFragment.java

```diff
@@ -190,6 +198,32 @@ public boolean onActionItemClicked(ActionMode actionMode, MenuItem menuItem) {
 
 		return true;
 	}
+	@RequiresApi(api = Build.VERSION_CODES.M)
+		private void checkPermissions(){
+			int permissionCheck = checkSelfPermission(getActivity().getApplicationContext(), Manifest.permission.WRITE_EXTERNAL_STORAGE);
+			if (!(permissionCheck == PackageManager.PERMISSION_GRANTED)) {
+					// User may have declined earlier, ask Android if we should show him a reason
+							if (shouldShowRequestPermissionRationale(Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+										} else {
+											requestPermissions(new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, CALLBACK_NUMBER);
+								}
+				} else {
+					startFileSharing(getCheckedFile());
+				}
+		}
+
+				@Override
```

Comment:
does this line need indentation also? it's in the middle of the code, and I suspect that's not too clean.

I have seen this: `@Override public boolean ....`